merge_global_stats.py: error: the following arguments are required: batch.global.stats

Config:

```yaml
########################################################################################################################
# input configuration
# This parameter denotes the input dir, a path to a directory with a very specific structure, that contains the input
#     data to this pipeline, i.e. the batches.
# Here we describe this structure in details.
# Each batch named <batch> is composed of one required file (<batch>.txt) and one optional file (<batch>.nw).
#   * The <batch.txt> file is a text file where each line is a path to an assembly file;
#      * An assembly file is a file with extensions ["fa", "fasta", "fna", "ffa"], optionally compressed with gzip
#          ("gz" extension) denoting the assembly of a sample;
#      * Every *.txt file in the input dir will be interpreted as a batch;
#      * See .test/input/test_run_with_tree.txt for an example of a <batch>.txt file;
#   * The <batch>.nw is an optional newick file representing the tree of the samples in its respective <batch>.txt
#          file. If this file is not provided, this pipeline builds a tree automatically using Mashtree.
#   * These files must be at the top level of the given directory.
input_dir: input
output_dir: output
########################################################################################################################


########################################################################################################################
# output configuration
generate_assembly_stats: True  # True or False
generate_prepropagation_stats: False  # True or False
generate_postpropagation_stats: False  # True or False
########################################################################################################################
```



```bash
$ m
snakemake -j --use-conda --conda-prefix="/Users/karel/github/my/mof-compress/conda" -p --rerun-incomplete
Building DAG of jobs...
Your conda installation is not configured to use strict channel priorities. This is however crucial for having robust and correct environments (for details, see https://conda-forge.org/docs/user/tipsandtricks.html). Please consider to configure strict priorities by executing 'conda config --set channel_priority strict'.
Creating conda environment workflow/envs/env.yaml...
Downloading and installing remote packages.
Environment for /Users/karel/github/my/mof-compress/workflow/rules/../envs/env.yaml created (location: conda/3d2d176b08a25d5ce8f7d00bb7fcdf15_)
Using shell: /Users/karel/miniconda/bin/bash
Provided cores: 10
Rules claiming more threads will be scaled down.
Job stats:
job             count
------------  -------
all                 1
global_stats        1
total               2

Select jobs to execute...

[Wed Oct 25 18:57:35 2023]
rule global_stats:
    output: output/global_stats.tsv
    jobid: 1
    reason: Missing output files: output/global_stats.tsv
    resources: tmpdir=/var/folders/z6/8lsjsl3x5q94j7041rtdwqyh0000gq/T


        /Users/karel/github/my/mof-compress/workflow/rules/../scripts/merge_global_stats.py  > output/global_stats.tsv
        
Activating conda environment: conda/3d2d176b08a25d5ce8f7d00bb7fcdf15_
usage: merge_global_stats.py [-h] batch.global.stats [batch.global.stats ...]
merge_global_stats.py: error: the following arguments are required: batch.global.stats
[Wed Oct 25 18:58:05 2023]
Error in rule global_stats:
    jobid: 1
    output: output/global_stats.tsv
    conda-env: /Users/karel/github/my/mof-compress/conda/3d2d176b08a25d5ce8f7d00bb7fcdf15_
    shell:
        
        /Users/karel/github/my/mof-compress/workflow/rules/../scripts/merge_global_stats.py  > output/global_stats.tsv
        
        (one of the commands exited with non-zero exit code; note that snakemake uses bash strict mode!)

Removing output files of failed job global_stats since they might be corrupted:
output/global_stats.tsv
Shutting down, this might take some time.
Exiting because a job execution failed. Look above for error message
Complete log: .snakemake/log/2023-10-25T185240.943802.snakemake.log
make: *** [all] Error 1
```

--- a/workflow/Snakefile
+++ b/workflow/Snakefile
@@ -14,10 +14,17 @@ report: "report/workflow.rst"
 include: "rules/common.smk"
 
 
+## WILDCARDS CONSTRAINS
+wildcard_constraints:
+    sample=r"[a-zA-Z0-9_-]+",
+    batch=r"[a-zA-Z0-9_-]+",
+    protocol=r"(asm|pre|post)",
+
+
+# TODO: add genome stats, make it configurable
 rule all:
     input:
         fn_stats_batches(),
-        # TODO: add genome stats, make it configurable
         #fn_stats_genomes(),
 
 

--- a/workflow/rules/common.smk
+++ b/workflow/rules/common.smk
@@ -55,33 +55,13 @@ for x in res:
                 sample = _get_sample_from_fn(sample_fn)
                 BATCHES_FN[batch][sample] = sample_fn
 
-
-## WILDCARDS CONSTRAINS
-wildcard_constraints:
-    sample=r"[a-zA-Z0-9_-]+",
-    batch=r"[a-zA-Z0-9_-]+",
-    protocol=r"(asm|pre|post)",
+assert len(BATCHES_FN)!=0, "\nERROR: No input files provided. Please provide at least one batch in '{dir_input()}/'.\n"
 
 
 ## BATCHES
 
-
 def get_batches():
     return BATCHES_FN.keys()
-
-
-## DIR PATHS
-
-
-def dir_prophyle(_batch):
-    return f"{dir_intermediate()}/post/{_batch}"
-
-
-def dir_prophyle_propagation(_batch):
-    return f"{dir_intermediate()}/post/{_batch}/propagation"
-
-
-## FILE PATHS
 
 
 #####################################
@@ -188,6 +168,15 @@ def fn_post_seq0(_batch, _node):
 
 def fn_post_seq(_batch, _node):
     return f"{dir_intermediate()}/post/{_batch}/{_node}.simpl"
+
+
+def dir_prophyle(_batch):
+    return f"{dir_intermediate()}/post/{_batch}"
+
+
+def dir_prophyle_propagation(_batch):
+    return f"{dir_intermediate()}/post/{_batch}/propagation"
+
 
 
 ## WILDCARD FUNCTIONS

